rust/datafusion/src/logical_plan/plan.rs

```diff
@@ -561,6 +561,7 @@ impl From<&PlanType> for String {
 
 /// Represents some sort of execution plan, in String form
 #[derive(Debug, Clone, PartialEq)]
+#[allow(clippy::rc_buffer)]
```

Comment:
Clippy is complaining about https://rust-lang.github.io/rust-clippy/master/index.html#rc_buffer - but I think in this case using the Arc<String> is really about cheaply copying the `String` rather than it being mutable. The same happens for the other places I disabled this lint in this PR

I am not enough of a Rust expert to know if there is some better way to pass around a cheaply cloneable `Vec` or `String` that would appease Clippy.

Reply:
Did it give an alternative in the suggestion?
 
Looks like `Arc<str>` and `Arc<[T]>` should be the alternatives

Reply:
I agree @Dandandan . I changed then a couple of times, but since the focus of this PR was the CI in stable, which includes build, tests, etc. I considered clippy to be a minor hassle that we can deal with later.

Reply:
@Dandandan  -- It did give alternatives but I did not figure out how to apply it.  

The actually clippy lint didn't make a whole lot of sense to me as it seems to be focused on the mutability of the Arc/Rc's elements, which doesn't seem to apply here. From my reading of this code and the other callsites, the use case of using `Arc<String>` is not *mutability* of the contained `String` but rather cheaply copying it around.

Using `Arc<str>` doesn't make sense as the actual String is allocated at runtime (it is the formatted version of an explain plan)

Reply:
There is some discussion about why it can make sense to use `Arc<str>` over `Arc<String>`:
https://users.rust-lang.org/t/arc-string-vs-arc-str/30571

of course, difference is minimal, only saving a bit of indirection/some bytes.

You can use `into` to convert a `String` or `&str` into a `Arc<str>` or `Rc<str>` 
https://doc.rust-lang.org/std/sync/struct.Arc.html#impl-From%3CString%3E

Example:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=eb6db938ae03b001edca59aba1512d33